src/segment_path.go

```diff
@@ -180,13 +190,12 @@ func (pt *path) rootLocation() string {
 
 func (pt *path) pathDepth(pwd string) int {
 	splitted := strings.Split(pwd, pt.env.getPathSeperator())
-	var validParts []string
+	depth := 0
 	for _, part := range splitted {
 		if part != "" {
-			validParts = append(validParts, part)
+			depth++
 		}
 	}
-	depth := len(validParts)
 	return depth - 1
 }
```

Comment:
Small performance improvement here.  I wrote a quick benchmark to prove it:

```

goos: windows
goarch: amd64
pkg: github.com/jandedobbeleer/oh-my-posh3
BenchmarkPathDepthOld-12           38396             31473 ns/op
BenchmarkPathDepthNew-12           40593             29583 ns/op
PASS
ok      github.com/jandedobbeleer/oh-my-posh3   3.968s


goos: linux
goarch: amd64
pkg: github.com/jandedobbeleer/oh-my-posh3
BenchmarkPathDepthOld-12           41817             27922 ns/op
BenchmarkPathDepthNew-12           45936             26325 ns/op
PASS
ok      github.com/jandedobbeleer/oh-my-posh3   3.021s
```

Benchmark code if you're interested:
```go
func BenchmarkPathDepthOld(b *testing.B) {
	for n := 0; n < b.N; n++ {
		pwd := "/usr"
		for i := 0; i < 99; i++ {
			pwd += levelDir
		}
		env := new(MockedEnvironment)
		env.On("getPathSeperator", nil).Return("/")
		path := &path{
			env: env,
		}
		path.pathDepthOld(pwd)
	}
}

func BenchmarkPathDepthNew(b *testing.B) {
	for n := 0; n < b.N; n++ {
		pwd := "/usr"
		for i := 0; i < 99; i++ {
			pwd += levelDir
		}
		env := new(MockedEnvironment)
		env.On("getPathSeperator", nil).Return("/")
		path := &path{
			env: env,
		}
		path.pathDepth(pwd)
	}
}
```